src/utils/dom-utils.ts

```diff
@@ -39,6 +39,24 @@ export function isEditableNode(node: Node | null): boolean {
   return isTextInputNode(node) || isContentEditableNode(node);
 }
 
+/**
+ * Tests whether an element is 'interactive', i.e. an element
+ * that we should not do lookups on when tapped on mobile.
+ */
+export function isInteractiveElement(elem: Element) {
+  return (
+    elem.tagName === 'A' ||
+    elem.tagName === 'BUTTON' ||
+    elem.tagName === 'INPUT' ||
+    elem.tagName === 'TEXTAREA' ||
+    elem.tagName === 'SELECT' ||
+    elem.tagName === 'DATALIST' ||
+    elem.tagName === 'OPTGROUP' ||
+    elem.tagName === 'OPTION' ||
+    isContentEditableNode(elem)
+  );
```

Comment:
Nit: If we make this take `Node | null` like many of the other functions in this file the call site will become a lot simpler.

```suggestion
export function isInteractiveElement(elem: Node | null) {
  return (
    isElement(elem) &&
    (elem.tagName === 'A' ||
      elem.tagName === 'BUTTON' ||
      elem.tagName === 'INPUT' ||
      elem.tagName === 'TEXTAREA' ||
      elem.tagName === 'SELECT' ||
      elem.tagName === 'DATALIST' ||
      elem.tagName === 'OPTGROUP' ||
      elem.tagName === 'OPTION' ||
      isContentEditableNode(elem))
  );
```

Reply:
Yeah, agreed. Though I think we'll want to move the `isContentEditableNode` call up to, since, AFAIK, the event target could still be a non-element Node, like a text node, inside of a content editable. I think we'd still want to prevent a tap in those cases.

Though now that I type it out, I'm not sure if the browser will ever have text nodes as event targets?